[CI] IncrementalBulkIT testMultipleBulkPartsWithBackoff failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #4276 / ubuntu-2204_platform-support-unix](https://gradle-enterprise.elastic.co/s/ng6qqjdyhajuk)
- [elasticsearch-periodic-platform-support #4276 / oraclelinux-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/bcv7zzcy3lfgs)
- [elasticsearch-periodic-platform-support #4276 / debian-11_platform-support-unix](https://gradle-enterprise.elastic.co/s/svz5d6tc6lgkk)
- [elasticsearch-periodic-platform-support #4276 / ubuntu-1804_platform-support-unix](https://gradle-enterprise.elastic.co/s/xrmzguujcp5go)
- [elasticsearch-periodic-platform-support #4276 / rocky-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/zwp5ndeboekwu)
- [elasticsearch-pull-request #34711 / part-1](https://gradle-enterprise.elastic.co/s/sfk53nb7qpspi)

**Reproduction Line:**
```
./gradlew ":server:internalClusterTest" --tests "org.elasticsearch.action.bulk.IncrementalBulkIT.testMultipleBulkPartsWithBackoff" -Dtests.seed=EE649D3DAE2BBD76 -Dtests.locale=lij-IT -Dtests.timezone=EET -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testMultipleBulkPartsWithBackoff'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.action.bulk.IncrementalBulkIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: null
```

**Issue Reasons:**
- [main] 2 consecutive failures in test testMultipleBulkPartsWithBackoff
- [main] 6 failures in test testMultipleBulkPartsWithBackoff (3.1% fail rate in 196 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -354,9 +354,6 @@ tests:
 - class: org.elasticsearch.xpack.inference.InferenceCrudIT
   method: testGet
   issue: https://github.com/elastic/elasticsearch/issues/114135
-- class: org.elasticsearch.action.bulk.IncrementalBulkIT
-  method: testIncrementalBulkHighWatermarkBackOff
-  issue: https://github.com/elastic/elasticsearch/issues/114073
 - class: org.elasticsearch.xpack.esql.expression.function.aggregate.AvgTests
   method: "testFold {TestCase=<double> #7}"
   issue: https://github.com/elastic/elasticsearch/issues/114175

--- a/server/src/main/java/org/elasticsearch/action/bulk/IncrementalBulkService.java
+++ b/server/src/main/java/org/elasticsearch/action/bulk/IncrementalBulkService.java
@@ -194,7 +194,7 @@ public class IncrementalBulkService {
                         releasables.clear();
                         // We do not need to set this back to false as this will be the last request.
                         bulkInProgress = true;
-                        client.bulk(bulkRequest, ActionListener.runAfter(new ActionListener<>() {
+                        client.bulk(bulkRequest, ActionListener.runBefore(new ActionListener<>() {
 
                             private final boolean isFirstRequest = incrementalRequestSubmitted == false;
 

